wp-content/themes/pub/wporg-learn-2024/inc/block-config.php

```diff
@@ -561,71 +561,39 @@ function inject_other_filters( $key ) {
 }
 
 /**
- * Modifies the courses query for the My Courses page, by adding courses to be excluded.
+ * Modify the course query on the 'My Courses' page to display courses according to the filter status.
+ * Corresponds to https://github.com/Automattic/sensei/blob/trunk/includes/blocks/course-list/class-sensei-course-list-student-course-filter.php#L95
  *
- * @param array $parsed_block The parsed block data.
- * @return array The modified block data.
+ * @param array $query The course query.
+ * @return array The modified course query.
  */
-function filter_student_course_list( $parsed_block ) {
+function modify_course_query( $query ) {
 	if (
-		'core/query' !== $parsed_block['blockName'] ||
-		'course' !== ( $parsed_block['attrs']['query']['postType'] ?? '' ) ||
-		get_the_ID() !== Sensei()->settings->get_my_courses_page_id()
+		'course' === $query['post_type'] &&
+		get_the_ID() === Sensei()->settings->get_my_courses_page_id()
```

Comment:
Not sure if you have any idea about this: I originally didn't include this, and in the course archive page, it didn't enter the condition, which I think it should have, as it uses the query loop block as well.

Reply:
This is still needed though because the front page will be affected. 
Maybe it's because of the query ID? The query ID for the front page and the 'My Courses' page is the same.

Reply:
And upon further thought, `'course' === $query['post_type']` can probably be removed because the query loop on this page only queries courses, what do you think?

Reply:
Yeah agreed, I think we can just check the page id

Reply:
fdad165